### PR TITLE
DRY'd startapp and startproject management commands

### PR DESCRIPTION
**Pull request by adamchainz**
Noticed this duplication and lack of test coverage whilst reading the code.

### CODE REVIEW

File: django/core/management/templates.py [hunk 1]
```diff
@@ -210,14 +211,35 @@ def handle_template(self, template, subdir):
...
+        a_or_an = 'an' if app_or_project == 'app' else 'a'
         if name is None:
-            raise CommandError("you must provide %s %s name" % (
-                "an" if app_or_project == "app" else "a", app_or_project))
-        # If it's not a valid directory name.
+            raise CommandError('you must provide {an} {app} name'.format(
+                an=a_or_an,
+                app=app_or_project,
+            ))
+        # Check it's a valid directory name
```
Comment (timgraham) [inline django/core/management/templates.py]: Use periods

File: tests/admin_scripts/tests.py [hunk 2]
```diff
@@ -1900,11 +1900,27 @@ def test_invalid_project_name(self):
...
                 err,
-                "Error: '%s' is not a valid project name. Please make "
-                "sure the name is a valid identifier." % bad_name
+                "CommandError: '{}' is not a valid project name. Please make "
+                "sure the name is a valid identifier.".format(bad_name)
             )
             self.assertFalse(os.path.exists(testproject_dir))
 
+    def test_importable_project_name(self):
+        "Make sure the startproject management command validates a project name"
```
Comment (timgraham) [inline tests/admin_scripts/tests.py]: Chop "Make sure" per https://code.djangoproject.com/ticket/27392

**Comment by adamchainz** [comment]
Done, I checked the tests do run separate to the behaviour change too 😉 

Guess all the test docstrings need changing in that file as they all have "make sure"

**Comment by timgraham** [comment]
Merged in b3e55109bd2ec5125042b65cdaac0789574c1848 and c61d1361d027a729d07d277879950ff133c19f4c.

**Pull request closed without merging**